### DISCUSSION TITLE
How to resolve error for this video and audio sync error

### DISCUSSION THREAD

**Discussion by jocelynasp** [bonsai-rx/bonsai#904] [Q&A]
I was looking in the forums for something related to recording both video and audio. We currently have four cameras (two in one computer, two in the other) running at once (reward training with a tone) and would like to integrate a microphone in the mix to match when the tone goes off when we later look at videos. 

I found the attached workflow which would be perfect for what we're doing, however when I try running it, I get the following error: "The system cannot find the file specified" 
I tried giving it a path to see if that would help, but I don't really know what to do at this point. 

![image](https://user-images.githubusercontent.com/99834393/168500937-a7851790-abfd-44d5-a45b-c48a66ac6566.png)

![image](https://user-images.githubusercontent.com/99834393/168500959-d0029b7e-8cff-422f-ac24-2d1e3311c7a0.png)
[ffmpeg-audio-video-sync.zip](https://github.com/bonsai-rx/bonsai/files/8696376/ffmpeg-audio-video-sync.zip)

I attached the workflow as well.

Thank you!

**Comment by bruno-f-cruz** [comment]
I just tried the workflow in my computer and it is working (albeit no sound in the video, but i am not familiar with this ffmpeg codec configuration). Can you try increasing the `DueTime` property in `DelaySubscription`? I wonder if your data streams are taking longer than mine to initialize, somehow.

**Reply by jocelynasp** [reply]
I have no idea what ffmpeg is supposed to be or do, I just snagged that from one of the Google Groups threads and it seemed to be what I need. I just don't know how to make it work. I just installed a package that has that name and it made no difference. I also tried your suggestion and am getting the same error. 

Do you by any chance know about assembling a workflow that can record video and audio and have it sync it? I'm recording from a camera that has no audio so I'm using an additional microphone.

**Reply by bruno-f-cruz** [reply]
Ok, I wonder if you tried to download FFMPEG before running the workflow. 
Here are the steps to set in up:
1. [Download the FFMPEG executable](https://ffmpeg.org/download.html#build-windows)
2. Place the `ffmpeg.exe` executable inside your Bonsai's extensions folder. E.g. `...\AppData\Local\Bonsai\Extensions`  (you can also manually point your `StartProcess` node to the  `ffmpeg.exe`)
3. In the `StartPrrocess` node, set your `FileName` property to `ffmpeg` (or point it to the ffmpeg executable manually)

Can you try these steps and let me know if it works?

Additionally, the lack of sound I described previously was due to my mic being off (...). Also, I noticed that in my computer, the red and blue channels were swapped in the output video. In order to fix it i changed the FFMPEG `-pix_fmt rgb24` to `-pix_fmt bgr24`. Something to keep in mind if you come across the same problem.

Hope it helps,
Bruno
